### PR TITLE
Fix: tunnel manager & tracker race condition

### PR DESCRIPTION
**Pull request by xjasonlyu**
fix https://github.com/Dreamacro/clash/issues/1041 by using `go.uber.org/atomic`

### CODE REVIEW

**Comment by Dreamacro** [comment]
To use `go.uber.org/atomic`, you can replace all atomic with the new `atomic`

**Comment by xjasonlyu** [comment]
> To use `go.uber.org/atomic`, you can replace all atomic with the new `atomic`

Done👌

File: adapters/outbound/base.go [hunk 1]
```diff
@@ -160,9 +161,9 @@ func (p *Proxy) MarshalJSON() ([]byte, error) {
 func (p *Proxy) URLTest(ctx context.Context, url string) (t uint16, err error) {
 	defer func() {
 		if err == nil {
```
Comment (Dreamacro) [inline adapters/outbound/base.go:163]: p.alive.Store(err == nil) 🤣

Comment (xjasonlyu) [reply adapters/outbound/base.go:163]: 光改代码忘记看逻辑了🌚

**Comment by Dreamacro** [comment]
LGTM, thanks

**Pull request merged**